lib/frontman/commands/serve.rb

```diff
@@ -83,8 +84,34 @@ def serve
   end
 end
 
-class FrontManServer < Sinatra::Base
-  set :port, 4568
+class FrontmanServer < Sinatra::Base
+  port = Frontman::Config.get(:port, fallback: 4568)
+  num_retries = Frontman::Config.get(:port_retries, fallback: 3)
+
+  port_retry_strategy = Frontman::Config.get(:port_retry_strategy, ->(port_num){
+    port_in_use = false
+
+    (1 + num_retries).times do
+      begin
+        port_in_use = Socket.tcp("localhost", port_num, connect_timeout: 3) { true }
+      rescue SocketError
```

Comment:
I get an error locally when there's no connection on the port we're checking. I believe it's the `socket` library that throws the system error, instead of wrapping their own. If we replace this line with the following, it works:

```suggestion
      rescue StandardError
```